### PR TITLE
Grouping parameters

### PR DESCRIPTION
**Pull request by mikolak**
Groundwork for the parameter paging feature. Contains logic for converting list of parameters to domain groups and creation of the tree, and ugly little view to make sure it's all working.

### CODE REVIEW

File: cockpit/src/main/res/drawable/chevron_right.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (martawoldanska) [inline cockpit/src/main/res/drawable/chevron_right.xml]: 👍

File: sample/cockpit/cockpit.yml [hunk 2]
```diff
@@ -40,6 +40,11 @@ showFooter:
   value: true
   group: "Footer"
 
+exampleSubGroupParam:
+  description: "Whatever really"
+  value: 500
+  group: "Footer/subgroup"
```
Comment (martawoldanska) [inline sample/cockpit/cockpit.yml]: And here as well :)

File: cockpit/src/main/java/com/polidea/cockpit/paramsedition/ParamsEditionAdapter.kt [hunk 3]
```diff
@@ -71,6 +73,9 @@ internal class ParamsEditionAdapter(var presenter: ParamsEditionContract.Present
         if (itemPosition.isGroupPosition())
```
Comment (mikolak) [inline cockpit/src/main/java/com/polidea/cockpit/paramsedition/ParamsEditionAdapter.kt]: Rename to `section`?

File: cockpit/src/main/java/com/polidea/cockpit/extensions/CockpitParamGroupExt.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    forEach {
+        val group = extractedSubgroups.getOrCreateGroupAndParents(it.key)
+        group.params.addAll(it.value)
+        extractedSubgroups[it.key] = group
+    }
+
+    return extractedSubgroups.toReducedNonMutable()
+}
+
+private fun MutableMap<String?, MutableCockpitParamGroup>.getOrCreateGroupAndParents(name: String?): MutableCockpitParamGroup {
```
[the remark this answers is not in the document]
Comment (mikolak) [reply cockpit/src/main/java/com/polidea/cockpit/extensions/CockpitParamGroupExt.kt]: Perhaps `getOrCreateGroupHierarchy()`?

**Pull request merged**